nil pointer on bgp

When creating a bgp connection, the resource is created but the provider crash just after:

```terraform
resource "routeros_bgp_connection" "srv1_cilium" {
  name         = "srv1_cilium"
  as           = "65000"
  remote {
    address = "192.168.40.41"
    as      = "65001"
  }
  local {
    role = "ebgp"
  }
}
```

```
routeros_bgp_connection.srv1_cilium: Creating...
╷
│ Error: Request cancelled
│
│   with routeros_bgp_connection.srv1_cilium,
│   on routing_bgp.tf line 30, in resource "routeros_bgp_connection" "srv1_cilium":
│   30: resource "routeros_bgp_connection" "srv1_cilium" {
│
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-routeros_v1.7.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xad4928]

goroutine 8 [running]:
github.com/terraform-routeros/terraform-provider-routeros/routeros.MikrotikResourceDataToTerraform(0xc00055f900?, 0xc00012e0a0?, 0x3?)
        github.com/terraform-routeros/terraform-provider-routeros/routeros/mikrotik_serialize.go:381 +0x15a8
github.com/terraform-routeros/terraform-provider-routeros/routeros.ResourceCreate({0xdbb408, 0xc0001309c0}, 0x12763e0?, 0x1176592e000?, {0xc44720?, 0xc0003d85f0})
        github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:90 +0x4f0
github.com/terraform-routeros/terraform-provider-routeros/routeros.DefaultCreate.func1({0xdbb408?, 0xc0001309c0?}, 0x0?, {0xc44720?, 0xc0003d85f0?})
        github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:212 +0x3d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0000b2000, {0xdbb440, 0xc000159260}, 0xd?, {0xc44720, 0xc0003d85f0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000b2000, {0xdbb440, 0xc000159260}, 0xc0005669c0, 0xc00055f780, {0xc44720, 0xc0003d85f0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000010060, {0xdbb440?, 0xc000159140?}, 0xc00062e280)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000000000, {0xdbb440?, 0xc0001584b0?}, 0xc000154070)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xc4a400?, 0xc000000000}, {0xdbb440, 0xc0001584b0}, 0xc000154000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002c41e0, {0xdbef60, 0xc000007a00}, 0xc000310000, 0xc000120510, 0x1267180, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0002c41e0, {0xdbef60, 0xc000007a00}, 0xc000310000, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-routeros_v1.7.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

```
  # routeros_bgp_connection.srv1_cilium will be created
  + resource "routeros_bgp_connection" "srv1_cilium" {
      + add_path_out     = "none"
      + address_families = "ip"
      + as               = "65000"
      + connect          = true
      + disabled         = false
      + hold_time        = (known after apply)
      + id               = (known after apply)
      + inactive         = (known after apply)
      + keepalive_time   = "3m"
      + listen           = true
      + name             = "srv1_cilium"
      + nexthop_choice   = "default"
      + routing_table    = (known after apply)
      + vrf              = "main"

      + local {
          + port = 179
          + role = "ebgp"
        }

      + remote {
          + address = "192.168.40.41"
          + as      = "65001"
          + port    = 179
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
2023-05-15T20:59:54.761+0200 [DEBUG] command: asking for input: "\nDo you want to perform these actions?"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

2023-05-15T20:59:56.760+0200 [INFO]  backend/local: apply calling Apply
2023-05-15T20:59:56.761+0200 [DEBUG] Building and walking apply graph for NormalMode plan
2023-05-15T20:59:56.761+0200 [DEBUG] Resource state not found for node "routeros_bgp_connection.srv1_cilium", instance routeros_bgp_connection.srv1_cilium
2023-05-15T20:59:56.761+0200 [DEBUG] ProviderTransformer: "routeros_bgp_connection.srv1_cilium (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/terraform-routeros/routeros"]
2023-05-15T20:59:56.761+0200 [DEBUG] ProviderTransformer: "data.sops_file.secrets (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/carlpett/sops"]
2023-05-15T20:59:56.761+0200 [DEBUG] ProviderTransformer: "routeros_bgp_connection.srv1_cilium" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/terraform-routeros/routeros"]
2023-05-15T20:59:56.762+0200 [DEBUG] ReferenceTransformer: "routeros_bgp_connection.srv1_cilium" references: []
2023-05-15T20:59:56.762+0200 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/carlpett/sops\"]" references: []
2023-05-15T20:59:56.762+0200 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/terraform-routeros/routeros\"]" references: [data.sops_file.secrets (expand)]
2023-05-15T20:59:56.762+0200 [DEBUG] ReferenceTransformer: "routeros_bgp_connection.srv1_cilium (expand)" references: []
2023-05-15T20:59:56.762+0200 [DEBUG] ReferenceTransformer: "data.sops_file.secrets (expand)" references: []
2023-05-15T20:59:56.762+0200 [DEBUG] pruneUnusedNodes: data.sops_file.secrets (expand) is no longer needed, removing
2023-05-15T20:59:56.762+0200 [DEBUG] pruneUnusedNodes: provider["registry.terraform.io/carlpett/sops"] is no longer needed, removing
2023-05-15T20:59:56.763+0200 [DEBUG] Starting graph walk: walkApply
2023-05-15T20:59:56.763+0200 [DEBUG] created provider logger: level=debug
2023-05-15T20:59:56.763+0200 [INFO]  provider: configuring client automatic mTLS
2023-05-15T20:59:56.771+0200 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/terraform-routeros/routeros/1.7.0/linux_amd64/terraform-provider-routeros_v1.7.0 args=[.terraform/providers/registry.terraform.io/terraform-routeros/routeros/1.7.0/linux_amd64/terraform-provider-routeros_v1.7.0]
2023-05-15T20:59:56.771+0200 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/terraform-routeros/routeros/1.7.0/linux_amd64/terraform-provider-routeros_v1.7.0 pid=1351222
2023-05-15T20:59:56.771+0200 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/terraform-routeros/routeros/1.7.0/linux_amd64/terraform-provider-routeros_v1.7.0
2023-05-15T20:59:56.776+0200 [INFO]  provider.terraform-provider-routeros_v1.7.0: configuring server automatic mTLS: timestamp=2023-05-15T20:59:56.776+0200
2023-05-15T20:59:56.788+0200 [DEBUG] provider: using plugin: version=5
2023-05-15T20:59:56.788+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: plugin address: address=/tmp/plugin2314918809 network=unix timestamp=2023-05-15T20:59:56.788+0200
2023-05-15T20:59:56.811+0200 [WARN]  Provider "registry.terraform.io/terraform-routeros/routeros" produced an invalid plan for routeros_bgp_connection.srv1_cilium, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .vrf: planned value cty.StringVal("main") for a non-computed attribute
      - .keepalive_time: planned value cty.StringVal("3m") for a non-computed attribute
      - .listen: planned value cty.True for a non-computed attribute
      - .nexthop_choice: planned value cty.StringVal("default") for a non-computed attribute
      - .add_path_out: planned value cty.StringVal("none") for a non-computed attribute
      - .connect: planned value cty.True for a non-computed attribute
      - .disabled: planned value cty.False for a non-computed attribute
      - .address_families: planned value cty.StringVal("ip") for a non-computed attribute
      - .local[0].port: planned value cty.NumberIntVal(179) for a non-computed attribute
      - .remote[0].port: planned value cty.NumberIntVal(179) for a non-computed attribute
routeros_bgp_connection.srv1_cilium: Creating...
2023-05-15T20:59:56.811+0200 [INFO]  Starting apply for routeros_bgp_connection.srv1_cilium
2023-05-15T20:59:56.811+0200 [DEBUG] routeros_bgp_connection.srv1_cilium: applying the planned Create change
2023-05-15T20:59:56.812+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: request body:  {"add-path-out":"none","address-families":"ip","as":"65000","connect":"yes","disabled":"no","keepalive-time":"3m","listen":"yes","local.port":"179","local.role":"ebgp","name":"srv1_cilium","nexthop-choice":"default","remote.address":"192.168.40.41","remote.as":"65001","remote.port":"179","vrf":"main"}: @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=851314be-84c2-64c5-b794-e654866fddf0 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:16 timestamp=2023-05-15T20:59:56.812+0200
2023-05-15T20:59:56.812+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: PUT request URL:  https://192.168.88.1/rest/routing/bgp/connection: @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=851314be-84c2-64c5-b794-e654866fddf0 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:16 timestamp=2023-05-15T20:59:56.812+0200
2023-05-15T20:59:57.241+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: response body: {".id":"*2","add-path-out":"none","address-families":"ip","as":"65000","connect":"true","disabled":"false","inactive":"false","keepalive-time":"3m","listen":"true","local.default-address":"192.168.40.21","local.port":"179","local.role":"ebgp","name":"srv1_cilium","nexthop-choice":"default","remote.address":"192.168.40.41","remote.as":"65001","remote.port":"179","routing-table":"main","vrf":"main"}: @module=routeros tf_provider_addr=terraform-routeros/routeros tf_req_id=851314be-84c2-64c5-b794-e654866fddf0 tf_rpc=Configure @caller=github.com/terraform-routeros/terraform-provider-routeros/routeros/log.go:16 timestamp=2023-05-15T20:59:57.241+0200
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: panic: runtime error: invalid memory address or nil pointer dereference
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xad4928]
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: goroutine 97 [running]:
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/terraform-routeros/terraform-provider-routeros/routeros.MikrotikResourceDataToTerraform(0xc000616c00?, 0xc00024c238?, 0x2?)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/terraform-routeros/terraform-provider-routeros/routeros/mikrotik_serialize.go:381 +0x15a8
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/terraform-routeros/terraform-provider-routeros/routeros.ResourceCreate({0xdbb408, 0xc00004d7a0}, 0x12763e0?, 0x1176592e000?, {0xc44720?, 0xc000612730})
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:90 +0x4f0
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/terraform-routeros/terraform-provider-routeros/routeros.DefaultCreate.func1({0xdbb408?, 0xc00004d7a0?}, 0x0?, {0xc44720?, 0xc000612730?})
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:212 +0x3d
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000550620, {0xdbb440, 0xc000412db0}, 0xd?, {0xc44720, 0xc000612730})
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:707 +0x12e
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000550620, {0xdbb440, 0xc000412db0}, 0xc000338dd0, 0xc000616a80, {0xc44720, 0xc000612730})
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00029a048, {0xdbb440?, 0xc000412c90?}, 0xc0006123c0)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002ba000, {0xdbb440?, 0xc000412270?}, 0xc00057c310)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xc4a400?, 0xc0002ba000}, {0xdbb440, 0xc000412270}, 0xc00057c000, 0x0)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b61e0, {0xdbef60, 0xc000092820}, 0xc0002be000, 0xc0002802a0, 0x1267180, 0x0)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: google.golang.org/grpc.(*Server).handleStream(0xc0001b61e0, {0xdbef60, 0xc000092820}, 0xc0002be000, 0x0)
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: google.golang.org/grpc.(*Server).serveStreams.func1.2()
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       google.golang.org/grpc@v1.51.0/server.go:965 +0x98
2023-05-15T20:59:57.244+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-05-15T20:59:57.245+0200 [DEBUG] provider.terraform-provider-routeros_v1.7.0:       google.golang.org/grpc@v1.51.0/server.go:963 +0x28a
2023-05-15T20:59:57.247+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-05-15T20:59:57.247+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-05-15T20:59:57.247+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/terraform-routeros/routeros/1.7.0/linux_amd64/terraform-provider-routeros_v1.7.0 pid=1351222 error="exit status 2"
2023-05-15T20:59:57.248+0200 [ERROR] vertex "routeros_bgp_connection.srv1_cilium" error: Plugin did not respond
2023-05-15T20:59:57.248+0200 [DEBUG] states/remote: state read serial is: 1; serial is: 1
2023-05-15T20:59:57.248+0200 [DEBUG] states/remote: state read lineage is: b8d2a0f3-8818-4d2d-ff03-2fc8d328413c; lineage is: b8d2a0f3-8818-4d2d-ff03-2fc8d328413c
╷
│ Error: Plugin did not respond
│
│   with routeros_bgp_connection.srv1_cilium,
│   on routing_bgp.tf line 30, in resource "routeros_bgp_connection" "srv1_cilium":
│   30: resource "routeros_bgp_connection" "srv1_cilium" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The
│ plugin logs may contain more details.
╵

Stack trace from the terraform-provider-routeros_v1.7.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xad4928]

goroutine 97 [running]:
github.com/terraform-routeros/terraform-provider-routeros/routeros.MikrotikResourceDataToTerraform(0xc000616c00?, 0xc00024c238?, 0x2?)
        github.com/terraform-routeros/terraform-provider-routeros/routeros/mikrotik_serialize.go:381 +0x15a8
github.com/terraform-routeros/terraform-provider-routeros/routeros.ResourceCreate({0xdbb408, 0xc00004d7a0}, 0x12763e0?, 0x1176592e000?, {0xc44720?, 0xc000612730})
        github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:90 +0x4f0
github.com/terraform-routeros/terraform-provider-routeros/routeros.DefaultCreate.func1({0xdbb408?, 0xc00004d7a0?}, 0x0?, {0xc44720?, 0xc000612730?})
        github.com/terraform-routeros/terraform-provider-routeros/routeros/resource_default_actions.go:212 +0x3d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000550620, {0xdbb440, 0xc000412db0}, 0xd?, {0xc44720, 0xc000612730})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000550620, {0xdbb440, 0xc000412db0}, 0xc000338dd0, 0xc000616a80, {0xc44720, 0xc000612730})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00029a048, {0xdbb440?, 0xc000412c90?}, 0xc0006123c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002ba000, {0xdbb440?, 0xc000412270?}, 0xc00057c310)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xc4a400?, 0xc0002ba000}, {0xdbb440, 0xc000412270}, 0xc00057c000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b61e0, {0xdbef60, 0xc000092820}, 0xc0002be000, 0xc0002802a0, 0x1267180, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0001b61e0, {0xdbef60, 0xc000092820}, 0xc0002be000, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-routeros_v1.7.0 plugin crashed!


```

--- a/routeros/mikrotik_serialize.go
+++ b/routeros/mikrotik_serialize.go
@@ -214,6 +214,11 @@ func TerraformResourceDataToMikrotik(s map[string]*schema.Schema, d *schema.Reso
 					filedNameInState := fmt.Sprintf("%v.%v.%v", terraformSnakeName, 0, fieldName)
 					fieldSchema := terraformMetadata.Elem.(*schema.Resource).Schema[fieldName]
 
+					// Skip all read-only properties.
+					if fieldSchema.Computed && !fieldSchema.Optional {
+						continue
+					}
+
 					if fieldSchema.Optional && !d.HasChange(filedNameInState) &&
 						isEmpty(filedNameInState, fieldSchema, d, ctyList.GetAttr(fieldName)) {
 						continue
@@ -377,6 +382,16 @@ func MikrotikResourceDataToTerraform(item MikrotikItem, s map[string]*schema.Sch
 					err = d.Set(terraformSnakeName, l)
 				case *schema.Resource:
 					var v any
+
+					if _, ok := s[terraformSnakeName].Elem.(*schema.Resource).Schema[subFieldSnakeName]; !ok {
+						diags = append(diags, diag.Diagnostic{
+							Severity: diag.Warning,
+							Summary:  "Field '" + terraformSnakeName + "." + subFieldSnakeName + "' not found in the schema",
+							Detail: fmt.Sprintf("[MikrotikResourceDataToTerraformDatasource] the datasource Schema sub-field was lost during development: ▷ '%s.%s' ◁",
+								terraformSnakeName, subFieldSnakeName),
+						})
+						continue
+					}
 
 					switch s[terraformSnakeName].Elem.(*schema.Resource).Schema[subFieldSnakeName].Type {
 					case schema.TypeString:

--- a/routeros/resource_bgp_connection.go
+++ b/routeros/resource_bgp_connection.go
@@ -278,6 +278,11 @@ func ResourceRoutingBGPConnection() *schema.Resource {
 						Optional:    true,
 						Description: "Local connection IPv4/6 address.",
 					},
+					"default_address": {
+						Type:        schema.TypeString,
+						Computed:    true,
+						Description: "",
+					},
 					"port": {
 						Type:         schema.TypeInt,
 						Optional:     true,

